smalltalksrc/VMMaker/DruidJIT.class.st

```diff
@@ -2483,14 +2473,846 @@ DruidJIT >> gen_ExtSendSuperBytecode [
+	self ArithmeticShiftRightCq: 29 R: t1.
+	self AndCq: 1 R: t1.
+	self CmpCq: 0 R: t1.
+	jump10 := self JumpZero: 0.
+	self AddCq: ((byte1 + (r2 << 8)) << 3) R: t2.
+	self MoveR: t0 M64: 8 r: t2.
+	jump2 := self Jump: 0.
+	currentBlock := self Label.
+	jump10 jmpTarget: currentBlock.
+	self MoveR: t2 R: t1.
+	self AndCq: 7 R: t1.
+	self CmpCq: 0 R: t1.
+	jump10 := self JumpNonZero: 0.
+	self MoveR: t2 R: t1.
+	self AndCq: objectMemory getMemoryMap getSpaceMaskToUse R: t1.
+	self CmpCq: objectMemory getMemoryMap getPermSpaceMask R: t1.
+	jumpTrue := self JumpZero: 0.
+	self MoveCq: 0 R: t1.
+	jumpNext := self Jump: 0.
+	jumpTrue jmpTarget: self Label.
+	self MoveCq: 1 R: t1.
+	jumpNext jmpTarget: self Label.
+	self CmpCq: 1 R: t1.
+	jump3 := self JumpNonZero: 0.
+	self MoveR: t0 R: t1.
+	self AndCq: objectMemory getMemoryMap getSpaceMaskToUse R: t1.
+	self CmpCq: objectMemory getMemoryMap getPermSpaceMask R: t1.
+	jump5 := self JumpZero: 0.
+	self MoveR: t2 R: TempReg.
+	backEnd saveAndRestoreLinkRegAround: [
+		self CallRT: ceStoreCheckTrampoline ].
+	jump1 := self Jump: 0.
+	currentBlock := self Label.
+	jump10 jmpTarget: currentBlock.
+	self MoveCq: 0 R: t1.
+	currentBlock := self Label.
+	jump3 jmpTarget: currentBlock.
+	currentBlock := self Label.
+	jump5 jmpTarget: currentBlock.
+	jump1 jmpTarget: currentBlock.
+	self AddCq: ((byte1 + (r2 << 8)) << 3) R: t2.
+	self MoveR: t0 M64: 8 r: t2.
+	jump1 := self Jump: 0.
+	currentBlock := self Label.
+	jump7 jmpTarget: currentBlock.
+	self AddCq: ((byte1 + (r2 << 8)) << 3) R: t2.
+	self MoveR: t0 M64: 8 r: t2.
+	currentBlock := self Label.
+	jump6 jmpTarget: currentBlock.
+	jump8 jmpTarget: currentBlock.
+	jump4 jmpTarget: currentBlock.
+	jump9 jmpTarget: currentBlock.
+	jump2 jmpTarget: currentBlock.
+	jump1 jmpTarget: currentBlock.
+	extA := 0.
+	^ 0
+]
+
+{ #category : #generated }
+DruidJIT >> gen_ExtStoreReceiverVariableBytecodeCommented [
+	"AutoGenerated by Druid"
+
+	| jump5 jump10 jump3 t3 jump8 jump1 tReceiver currentBlock jumpNext jump6 r2 live jump4 jumpTrue jump9 jump2 t2 jump7 tValue |
+	live := 0.
+	self annotateBytecode: self Label.
+	r2 := extA.
+	tValue := self
+		      allocateRegNotConflictingWith: live
+		      ifNone: [ ^ self unknownBytecode ].
+	live := live bitOr: (self registerMaskFor: tValue).
+	(self ssValue: 0) copyToReg: tValue.
+	self ensureReceiverResultRegContainsSelf.
+	tReceiver := self
+		      allocateRegNotConflictingWith: live
+		      ifNone: [ ^ self unknownBytecode ].
+	live := live bitOr: (self registerMaskFor: tReceiver).
+	self MoveR: ReceiverResultReg R: tReceiver.
+	t2 := self
+		      allocateRegNotConflictingWith: live
+		      ifNone: [ ^ self unknownBytecode ].
+	live := live bitOr: (self registerMaskFor: t2).
+	t3 := self
+		      allocateRegNotConflictingWith: live
+		      ifNone: [ ^ self unknownBytecode ].
+	live := live bitOr: (self registerMaskFor: t3).
+	
+	(byte1 + (r2 << 8))<=ReceiverIndex ifTrue: [
+		"Receiver is (not) context?"
```

Comment:
this one you did manually right?

Reply:
Hahaha ups, yes! I used to debug the code and find the bug :)

I'll remove it in the branch where I'm working now.